sdk/keyvault/Azure.Security.KeyVault.Certificates/src/LightweightPkcs8Decoder.cs

```diff
@@ -180,13 +171,62 @@ public ECDsa ToECDsa()
             public object ToObject()
             {
                 ECParameters ecParameters = new ECParameters();
-                ecParameters.Curve = ECCurve.CreateFromValue(Curve);
+                ecParameters.Curve = (ECCurve)Curve.ToObject();
                 ecParameters.D = D;
                 ecParameters.Q.X = X;
                 ecParameters.Q.Y = Y;
 
                 return ecParameters;
             }
         }
+
+        private struct ECCurveProxy
+        {
+            private static MethodInfo s_exportParametersMethod;
+            private static FieldInfo s_curveField;
+
+            private readonly string _namedCurveOid;
+            private readonly object _curve;
+
+            public ECCurveProxy(string namedCurveOid)
+            {
+                _namedCurveOid = namedCurveOid;
+                _curve = null;
+            }
+
+            private ECCurveProxy(object curve)
+            {
+                _namedCurveOid = null;
+                _curve = curve;
+            }
+
+            public static ECCurveProxy ExportFromPublicKey(ECDsa publicKey)
+            {
+                if (s_exportParametersMethod is null)
+                {
+                    s_exportParametersMethod = typeof(ECDsa).GetMethod("ExportParameters", BindingFlags.Public | BindingFlags.Instance, null, new[] { typeof(bool) }, null)
+                        ?? throw new PlatformNotSupportedException("The current platform does not support reading an ECDsa private key from a PEM file");
+
+                    s_curveField = s_exportParametersMethod.ReturnType.GetField("Curve", BindingFlags.Public | BindingFlags.Instance);
+                }
+
+                object parameters = s_exportParametersMethod.Invoke(publicKey, new object[] { false });
```

Comment:
You could make this array be a static readonly and avoid creating it once per call.